File: js/level19 Greedy/1541.js
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = require("fs").readFileSync(filePath).toString().trim();

let flag = true;
let ans = 0;
for(let i=0; i<input.length; i++){
  let num = '';
  while(i<input.length && input[i] !== '+' && input[i] !== '-'){
    num += input[i++];
  }
  if(flag)
    ans += parseInt(num);
  else
    ans -= parseInt(num);
  
  if(input[i] === '-')
  flag = false;
  }
console.log(ans)

File: baekjoon/cpp/level16/1541.cpp
#include<iostream>
#include<string>

using namespace std;

int main(){
    string equation;
    cin >> equation;

    bool flag = 1;
    int ans = 0;
    for(int i=0; i<equation.length(); i++){
        string num;
        while(i<equation.length() && equation[i] != '-' && equation[i] != '+'){
            num += equation[i++];
        }
        
        if(flag)
            ans += stoi(num);
        else
            ans -= stoi(num);
        
        
        if(flag && equation[i] == '-')
            flag = 0;
    }
    cout << ans;

        
}